src/psyclone/psyir/nodes/routine.py

```diff
@@ -0,0 +1,141 @@
+    :param bool entry_point: whether this Routine represents the entry point \
+                             into a program (i.e. Fortran Program or C main()).
+    :param return_type: the return-type of this routine.
+    :type return_type: :py:class:`psyclone.psyir.symbols.DataType` or NoneType
+    :param children: the PSyIR nodes that are children of this Routine.
+    :type children: list of :py:class:`psyclone.psyir.nodes.Node`
+    :param parent: the parent node of this Schedule in the PSyIR.
+    :type parent: :py:class:`psyclone.psyir.nodes.Node` or NoneType
+    :param symbol_table: initialise the Schedule with a given symbol table.
+    :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable` or \
+            NoneType
+
+    :raises TypeError: if the supplied routine name is not a str.
+    :raises TypeError: if the supplied entry_point is not a bool.
+    :raises TypeError: if the supplied return_type is not a DataType.
+
+    '''
+    # Textual description of the node.
+    _children_valid_format = "[Statement]*"
+    _text_name = "Schedule"
+    _colour_key = "Schedule"
+
+    def __init__(self, name, entry_point=False, return_type=None,
+                 children=None, parent=None, symbol_table=None):
+        super(Routine, self).__init__(children=children, parent=parent,
+                                      symbol_table=symbol_table)
+        self.name = name
+
+        if not isinstance(entry_point, bool):
+            raise TypeError("Routine 'entry_point' must be a bool but got "
+                            "'{0}'".format(type(entry_point).__name__))
+        self._entry_point = entry_point
+
+        if return_type and not isinstance(return_type, DataType):
+            raise TypeError("Routine 'return_type' must be of type DataType "
+                            "but got '{0}'".format(type(return_type).__name__))
+        self._return_type = return_type
+
+    @property
+    def dag_name(self):
+        '''
+        :returns: the name of this node in the dag.
+        :rtype: str
+        '''
+        return "_".join(["routine", self.name, str(self.abs_position)])
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of this Routine.
+        :rtype: str
+        '''
+        return self._name
+
+    @name.setter
+    def name(self, new_name):
+        '''
+        Sets a new name for the Routine.
+
+        :param str new_name: New name for the Routine.
+        '''
+        if not isinstance(new_name, str):
+            raise TypeError("Routine name must be a str but got "
+                            "'{0}'".format(type(new_name).__name__))
+        self._name = new_name
+
+    def __str__(self):
+        result = "{0}[{1}]:\n".format(self.coloured_name(False), self.name)
+        for entity in self._children:
+            result += str(entity) + "\n"
+        result += "End " + self.coloured_name(False)
+        return result
+
+    @property
+    def entry_point(self):
```

Comment:
I don't like the use of the name entry_point as I don't think it equates to a program. Would is_program or is_main work instead?

Reply:
I was trying hard to think of a language-agnostic name. I could go with either of your suggestions. What do @sergisiso and @hiker think?

Reply:
I've elected for "is_program". If anyone feels strongly they can shout :-)

Reply:
I'm happy :-)